File: 1089.复写零.py
#
# @lc app=leetcode.cn id=1089 lang=python3
#
# [1089] 复写零
#

# @lc code=start
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        possible_dups = 0
        length = len(arr) - 1

        for left in range(length + 1):
            if left > length - possible_dups:
                break
        
            if arr[left] == 0:
                if left == length - possible_dups:
                    arr[length] = 0
                    length -= 1
                    break
                possible_dups += 1
        last = length - possible_dups
        for i in range(last, -1, -1):
            if arr[i] == 0:
                arr[i + possible_dups] = 0
                possible_dups -= 1
                arr[i + possible_dups] = 0
            else:
                arr[i+possible_dups] = arr[i]

# @lc code=end

File: 1089.复写零.cpp
/*
 * @lc app=leetcode.cn id=1089 lang=cpp
 *
 * [1089] 复写零
 */

// @lc code=start
class Solution {
public:
    void duplicateZeros(vector<int>& arr) {
        vector<int>tmp;
        int pos = 0;
        while(tmp.size() < arr.size()){
            if(arr[pos] != 0){
                tmp.push_back(arr[pos]);
            }
            else{
                tmp.push_back(0);
                if(tmp.size() == arr.size())
                    break;
                tmp.push_back(0);
            }
            pos++;
        }
        arr = tmp;
        return ;
    }
};
// @lc code=end

